pkg/mpc-examples/contracts/01-weight-changer/WeightChangerController.sol

```diff
@@ -0,0 +1,164 @@
+    }
+
+    /**
+     * @dev Starts the gradual reweight process to bring the token's weights to 99/01.
+     * @dev Gradual reweight will start when this function is called and take _REWEIGHT_DURATION to complete.
+     */
+    function make9901() public {
+        uint256[] memory newWeights = new uint256[](2);
+        newWeights[0] = 99e16;
+        newWeights[1] = 1e16;
+        _updateWeights(block.timestamp, block.timestamp + _REWEIGHT_DURATION, getTokens(), newWeights);
+    }
+
+    // === Public Getters ===
+    function getMaximumWeight() public pure returns (uint256) {
+        return _MAX_WEIGHT;
+    }
+
+    function getMinimumWeight() public pure returns (uint256) {
+        return _MIN_WEIGHT;
+    }
+
+    function getCurrentWeights() public view returns (uint256[] memory) {
+        return _getPool().getNormalizedWeights();
+    }
+
+    function getReweightDuration() public pure returns (uint256) {
+        return _REWEIGHT_DURATION;
+    }
+
+    function getTokens() public view returns (IERC20[] memory) {
+        return _tokens;
+    }
+
+    function getPoolId() public view returns (bytes32) {
+        return _poolId;
+    }
+
+    function getVault() public view returns (IVault) {
+        return _vault;
+    }
+
+    /// === Private and Internal ===
+    function _getPoolFromId(bytes32 poolId) internal pure returns (IManagedPool) {
+        // 12 byte logical shift left to remove the nonce and specialization setting. We don't need to mask,
+        // since the logical shift already sets the upper bits to zero.
+        return IManagedPool(address(uint256(poolId) >> (12 * 8)));
+    }
+
+    function _getPool() internal view returns (IManagedPool) {
+        return _getPoolFromId(getPoolId());
+    }
+
+    function _setTokens(IERC20[] memory tokens) internal {
+        _tokens = ComposablePoolLib.dropBptFromTokens(tokens);
+    }
+
+    /**
+     * @dev Updates the weights of the managed pool.
+     * @param startTime The timestamp, in seconds, at when the gradual weight update process starts.
+     * @param endTime The timestamp, in seconds, at when the gradual weight update process is complete.
+     * @param tokens An array of tokens, IERC20, that make up the managed pool.
+     * @param weights The desired end weights of the pool tokens. Must correspond with the tokens parameter.
+     */
+    function _updateWeights(
+        uint256 startTime,
+        uint256 endTime,
+        IERC20[] memory tokens,
+        uint256[] memory weights
+    ) internal {
+        _verifyWeights(weights);
+        _getPool().updateWeightsGradually(startTime, endTime, tokens, weights);
+    }
+
+    function _verifyWeight(uint256 normalizedWeight) internal pure returns (uint256) {
+        require(normalizedWeight >= _MIN_WEIGHT, "Weight under minimum");
+        require(normalizedWeight <= _MAX_WEIGHT, "Weight over maximum");
+        return normalizedWeight;
+    }
```

Comment:
This is using the "returning unaltered input" pattern that had been flagged in the LiquidIndexController review. Are we fine with this? cc: @rabmarut

Reply:
I'd prefer to steer away from that pattern, even in this example code base. It would be better to either use the require statements directly or, if repeated frequently, rewrite the function such that it doesn't return anything. I would also also potentially advocate for a nomenclature change; I've never been 100% sold on "verify" - is there an industry standard nomenclature? They call it the CEI pattern, so maybe "check" is preferred?

Reply:
Yeah, I think having this would be fine:
```
_checkWeight(weight) 
<do something with weight>
```
with `_checkWeight(weight)` just having the two require statements and no return value.

Reply:
I also think it's worth putting all `check*` function invocations as close to the top of the local context as possible, to conform to the CEI pattern (though it doesn't seem strictly necessary).